### PR TITLE
Style updates to card course component

### PR DESCRIPTION
**Pull request by lmounsey**
# Description

This includes style updates to the `CardCourse` component, as well as alternative styling as a search result, which highlights matching text within the name. This util function will be used on another future component.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# Checklist:

- [x] Make sure to do not repeat yourself (DRY)
- [ ] Snapshots tested
- [ ] A11y tested
- [x] CSS utilises BEM naming convention and structure
- [ ] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [ ] Crossbrowser testing (IE11, Safari 8+, iOS 8.4+, Android 4.4+, Firefox ESR (v52.x), iPhone (4s,6), iPad 2, Galaxy s5) 
- [x] My changes generate no new warnings
- [ ] I have added tests (e2e and unit) that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Added the new component to the index.js (Vue and Lib) export

# Reviewer Checklist

- [ ] Code is not repeated (DRY)
- [ ] ES6+ code is used where possible
- [ ] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [ ] CSS utilises BEM naming convention and structure
- [ ] Snapshots tested against a copy of dev snapshots to check UI changes are intended
- [ ] A11y tested
- [ ] Crossbrowser tested in at least IE11

### CODE REVIEW

**Comment by gepifanio** [comment]
Comment: Deploy Started

Building with commit 8dc1dcf5d9c0cdf03aeed639c4946410511b0e59

https://app.netlify.com/sites/pattern-lib-unimelb/deploys/606ea0fee32b9c00079abc88

File: components/cards/_card-course.css [hunk 1]
```diff
@@ -2,22 +2,28 @@
...
   color: rgb(var(--col-brand));
   border: 1px solid rgb(var(--col-navy-mid-1));
+  border-radius: 2px;
 
   &::before {
     content: '';
-    top: 0;
-    left: 0;
+    /* stylelint-disable */
+    top: -1px;
```
Comment (gepifanio) [inline components/cards/_card-course.css:18]: please use rem instead of px

Comment (gepifanio) [inline components/cards/_card-course.css:18]: please use rem instead of px

File: components/cards/_card-course.css [hunk 2]
```diff
@@ -2,22 +2,28 @@
...
+    left: -1px;
     width: .389rem;
-    height: 100%;
+    height: calc(100% + 2px);
+    /* stylelint-enable */
     position: absolute;
     background-color: rgb(var(--col-black-light));
+    border-top-left-radius: 2px;
```
Comment (gepifanio) [inline components/cards/_card-course.css:25]: please use rem instead of px

**Review by gepifanio — CHANGES_REQUESTED** [review]
Hey @lmounsey, It looks good for me, just wondering if you could add a new variation in the storybook showing the card with the search query prop activating it, just to exemplify for the people how to use it, you could use the existing story to add a new card with the prop as well.

File: components/cards/_card-course.css (hunk 1 above)
Comment (lmounsey) [reply components/cards/_card-course.css]: Updated this to use rems

File: components/cards/_card-course.css (hunk 2 above)
Comment (lmounsey) [reply components/cards/_card-course.css]: Updated this to use rems

**Review by gepifanio — APPROVED** [review]
looks good just needs to fix the test

**Comment by lmounsey** [comment]
> looks good just needs to fix the test

Yep - test should be fixed now!

**Review by jaredh-um — APPROVED** [review]

**Pull request merged**